drivers/shared/executor/executor.go

```diff
@@ -577,6 +579,47 @@ func lookupBin(taskDir string, bin string) (string, error) {
 	return "", fmt.Errorf("binary %q could not be found", bin)
 }
 
+// lookupTaskBin finds the file in:
+// taskDir/local, taskDir, PATH search inside taskDir
+// and returns an absolute path
+func lookupTaskBin(taskDir string, bin string) (string, error) {
+	// Check in the local directory
+	localDir := filepath.Join(taskDir, allocdir.TaskLocal)
+	local := filepath.Join(localDir, bin)
+	if _, err := os.Stat(local); err == nil {
+		return local, nil
+	}
+
+	// Check at the root of the task's directory
+	root := filepath.Join(taskDir, bin)
+	if _, err := os.Stat(root); err == nil {
+		return root, nil
+	}
+
+	// Check the host PATH anchored inside the taskDir
+	if !strings.Contains(bin, "/") {
```

Comment:
This won't work on Windows. Windows uses `\`, but using [`os.PathSeparator`](https://golang.org/pkg/os/#PathSeparator) alone isn't enough because Windows also has the drive prefix (C:). Maybe use in combination with https://golang.org/pkg/path/filepath/#VolumeName

Reply:
In order to clarify this, I'm moving the new lookupTaskBin to executor_linux, which more accurately shows who calls it